src/factories/request.js

```diff
@@ -75,23 +97,13 @@ const fetchRetry = (
         if (response.ok) {
           resolve(response.json)
         }
-        if (attempt < config.fetchMaxAttempts && response.status === 429) {
-          setTimeout(
-            () =>
-              fetchRetry(
-                config,
-                uri,
-                method,
-                version,
-                headers,
-                requestBody,
-                attempt + 1
-              )
-                .then(result => resolve(result))
-                .catch(error => reject(error)),
-            attempt * config.retryDelay +
-              Math.floor(Math.random() * config.retryJitter)
-          )
+        if (attempt < config.fetchMaxAttempts) {
+          if (response.status === 401) {
+            console.log('retry')
+            this.authenticate().then(retryTimeout())
```

Comment:
```suggestion
           authenticate().then(retryTimeout())
```

doesn't need the this chaining here